components/image/child-components/figure.js

```diff
@@ -0,0 +1,25 @@
+import PropTypes from 'prop-types';
+import { StyledComponentContext } from '../../styled-components-context';
+import { InlineControlsStyleWrapper } from '../styles';
+
+export const Figure = (props) => {
+	const { style, children, ...rest } = props;
+
+	return (
+		<StyledComponentContext cacheKey="tenup-component-image">
+			<InlineControlsStyleWrapper {...style} {...rest}>
```

Comment:
@fabiankaegy Think this is where I got confused and am not sure I'm following 😆. Apologies! So, the `Figure` component already had the `figure` wrapper before the change:
```
const Figure = (props) => {
	const { children, style, ...rest } = props;

	return (
		<figure style={{ position: 'relative', ...style }} {...rest}>
			{children}
		</figure>
	);
};
```
I basically just swapped it for a styled component that also renders a figure.

Should there not be other conditional log in the main [component](https://github.com/10up/block-components/pull/269/files#diff-014d8ab1b491f70d246d8f0179a70772b6d2645a576e1e9859ac6c2c49a70783R104-R118)? Or am I missing something completely.

Reply:
@ncoetzer Okay I may have misunderstood the code then. 

What I was trying to get at is that the end markup should look like this: 

```html
<figure>
    <img />
    <div>
        <!-- Insert anything else here -->
    </div>
</figure>
````

and not:

```html
<figure>
    <div>
         <img />
        <!-- Insert anything else here -->
    </div>
</figure>
```

Reply:
@fabiankaegy Aah okay cool. In that case, yes, that is how it currently works and outputs the code. However, if someone were to use the component with this change, e.g.:
```
<Image id={someid} [any other props] />
```
it would still wrap the `<img>` tag inside a `<figure>`, which is a breaking change if I understood things correctly. So, I'm thinking of changing the logic in the main component to conditionally check if inline controls are needed, and if so, wrap the `<img>` in a `figure`. Otherwise, it will just output the `<img>` directly as it currently does. We could do this with any other feature placed behind a flag.

Of course, this new API also accommodates creating custom children, so the user would be able to override anything we have with `children`.

Thoughts?